converter/geant4/gdml/builder.py

```diff
@@ -0,0 +1,93 @@
+from typing import Optional
+# skipcq: BAN-B405
+import xml.etree.ElementTree as ET
+from defusedxml.minidom import parseString
+
+from converter.geant4 import utils
+from .materials import collect, create_material_elements
+from .structure import emit_structure
+
+
+def generate_gdml_entry_point(world_json: Optional[dict]) -> str:
+    """Generate GDML string for provided world or an empty world."""
+    if world_json:
+        return _generate_gdml(world_json)
+    return _generate_empty()
+
+
+def _generate_gdml(world: dict) -> str:
+    """Generate GDML from geometry node."""
+    gdml_root = ET.Element("gdml", {
+        "xmlns:xsi": "http://www.w3.org/2001/XMLSchema-instance",
+        "xsi:noNamespaceSchemaLocation":
+            "http://cern.ch/service-spi/app/releases/GDML/schema/gdml.xsd",
+    })
+
+    ET.SubElement(gdml_root, "define")
+
+    used_materials = collect(world)
```

Comment:
Can you rename `collect` to something more descriptive ? Even `get_materials` looks better